test/e2e/storage/non_graceful_node_shutdown.go

```diff
@@ -0,0 +1,186 @@
+
+	ginkgo.BeforeEach(func() {
+		c = f.ClientSet
+		ns = f.Namespace.Name
+	})
+
+	ginkgo.Describe("NonGracefulNodeShutdown", func() {
+		ginkgo.Context("pod that uses a persistent volume via gce pd driver", func() {
+			ginkgo.It("should get immediately rescheduled to a different node after non graceful node shutdown ", func() {
+				// Install gce pd csi driver
+				ginkgo.By("deploying csi gce-pd driver")
+				driver := drivers.InitGcePDCSIDriver()
+				config, cleanup := driver.PrepareTest(f)
+				dDriver, ok := driver.(storageframework.DynamicPVTestDriver)
+				if !ok {
+					e2eskipper.Skipf("csi driver expected DynamicPVTestDriver but got %v", driver)
+				}
+				defer cleanup()
+				ginkgo.By("Creating a gce-pd storage class")
+				sc := dDriver.GetDynamicProvisionStorageClass(config, "")
+				_, err := c.StorageV1().StorageClasses().Create(context.TODO(), sc, metav1.CreateOptions{})
+				framework.ExpectNoError(err)
+				scName := &sc.Name
+
+				ginkgo.By("Creating a pvc using the gce-pd storage class")
+				pvc := e2epv.MakePersistentVolumeClaim(e2epv.PersistentVolumeClaimConfig{
+					StorageClassName: scName,
+				}, ns)
+				gotPVC, err := c.CoreV1().PersistentVolumeClaims(ns).Create(context.TODO(), pvc, metav1.CreateOptions{})
+				framework.ExpectNoError(err)
+
+				ginkgo.By("Creating a deployment using the pvc " + pvc.Name)
+				podLabels := map[string]string{"app": "sts-pod-gcepd"}
+				dep := makeDeployment(ns, "sts-pod-gcepd", gotPVC.Name, podLabels)
+				_, err = c.AppsV1().Deployments(ns).Create(context.TODO(), dep, metav1.CreateOptions{})
+				framework.ExpectNoError(err)
+
+				ginkgo.By(fmt.Sprintf("Ensuring that the pod of deployment %s is running and ready", dep.Name))
+				labelSelector := labels.SelectorFromSet(labels.Set(podLabels))
+				podList, err := e2epod.WaitForPodsWithLabelRunningReady(c, ns, labelSelector, 1, framework.PodStartTimeout)
+				framework.ExpectNoError(err)
+				pod := &podList.Items[0]
+				oldNodeName := pod.Spec.NodeName
+
+				ginkgo.By("Stopping the kubelet non gracefully for pod" + pod.Name)
+				utils.KubeletCommand(utils.KStop, c, pod)
+
+				ginkgo.By("Adding out of service taint on node " + oldNodeName)
+				// taint this node as out-of-service node
+				taint := v1.Taint{
+					Key:    v1.TaintNodeOutOfService,
+					Effect: v1.TaintEffectNoExecute,
+				}
+				e2enode.AddOrUpdateTaintOnNode(c, oldNodeName, taint)
+
+				ginkgo.By(fmt.Sprintf("Checking if the pod %s got rescheduled to a new node", pod.Name))
+				labelSelectorStr := labels.SelectorFromSet(podLabels).String()
+				podListOpts := metav1.ListOptions{
+					LabelSelector: labelSelectorStr,
+					FieldSelector: fields.OneTermNotEqualSelector("spec.nodeName", oldNodeName).String(),
+				}
+				_, err = e2epod.WaitForPodsWithListOptionsRunningReady(c, podListOpts, ns, 1)
+				framework.ExpectNoError(err)
+				utils.KubeletCommand(utils.KStart, c, pod)
+				e2enode.RemoveTaintOffNode(c, oldNodeName, taint)
```

Comment:
We should verify that the node becomes ready for workloads after removing the taint. Can we check kubelet status 
 to become ready if possible? And / Or verify by creating a pod on this node or moving the existing pod back here?

Reply:
The method `utils.KubeletCommand` already wait for Kubelet to become ready. 
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/utils/pod.go#L161

Reply:
Can we also start a new pod on this node and check that it ends in running state? The test may take longer but it confirms that with the new behavior of the 'out-of-service' taint,  the node behaves as expected after the taint is removed.

Reply:
Does this https://github.com/kubernetes/kubernetes/pull/111380/files#diff-a99261da87dc4f2f476220e371c566b3e7ab9ec26634cb1097f4db2114c43a65R131 not address the comment. 
I have added this earlier after your review comment.

Reply:
```
createAndVerifyStatefulDeployment(scName, newDeploymentName, ns, newPodLabels, c)
```
@ Line 131 ( test/e2e/storage/non_graceful_node_shutdown.go )

Reply:
The change actually does. My bad, I  was seeing the older version of the code.

Reply:
No problem. This is a good one to have. Thanks for pointing this out.